test/e2e/specs/editor/blocks/gallery.spec.js

```diff
@@ -74,4 +83,160 @@ test.describe( 'Gallery', () => {
 			.poll( editor.getEditedPostContent )
 			.toBe( editedPostContent );
 	} );
+
+	test( 'can be created using uploaded images', async ( {
+		admin,
+		editor,
+		page,
+		galleryBlockUtils,
+	} ) => {
+		await admin.createNewPost();
+		await editor.insertBlock( { name: 'core/gallery' } );
+		const galleryBlock = page.locator(
+			'role=document[name="Block: Gallery"i]'
+		);
+		await expect( galleryBlock ).toBeVisible();
+
+		const filename = await galleryBlockUtils.upload(
+			galleryBlock.locator( 'data-testid=form-file-upload-input' )
+		);
+
+		const image = galleryBlock.locator( 'role=img' );
+		await expect( image ).toBeVisible();
+		await expect( image ).toHaveAttribute( 'src', new RegExp( filename ) );
+
+		const regex = new RegExp(
+			`<!-- wp:gallery {\\"linkTo\\":\\"none\\"} -->\\s*<figure class=\\"wp-block-gallery has-nested-images columns-default is-cropped\\"><!-- wp:image {\\"id\\":\\d+,\\"sizeSlug\\":\\"(?:full|large)\\",\\"linkDestination\\":\\"none\\"} -->\\s*<figure class=\\"wp-block-image (?:size-full|size-large)\\"><img src=\\"[^"]+\/${ filename }\.png\\" alt=\\"\\" class=\\"wp-image-\\d+\\"\/><\/figure>\\s*<!-- \/wp:image --><\/figure>\\s*<!-- \/wp:gallery -->`
+		);
+		await expect.poll( editor.getEditedPostContent ).toMatch( regex );
+	} );
+
+	test( 'gallery caption can be edited', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		const galleryCaption = 'Tested gallery caption';
+
+		await admin.createNewPost();
+		await editor.insertBlock( {
+			name: 'core/gallery',
+			innerBlocks: [
+				{
+					name: 'core/image',
+					attributes: {
+						id: uploadedMedia.id,
+						url: uploadedMedia.source_url,
+					},
+				},
+			],
+		} );
+
+		const gallery = page.locator( 'role=document[name="Block: Gallery"i]' );
+
+		await expect( gallery ).toBeVisible();
+		await editor.selectBlocks( gallery );
```

Comment:
I had to use `editor.selectBlocks()` util. Selecting a Gallery block using `page.click()` isn't reliable, it can easily select an inner image block. Using `